docs/specs/component.md

```diff
@@ -128,6 +128,61 @@ the reception of Vector events from an upstream component.
     defined properties as key-value pairs.
   - MUST NOT be rate limited.
 
+#### ComponentBytesReceived
+
+*Sources* MUST emit a `ComponentBytesReceived` event immediately after receiving, decompressing
+and filtering bytes from the upstream source and before the creation of a Vector event.
+
+- Properties
+  - `byte_size`
+    - For UDP, TCP, and Unix protocols, the total number of bytes received from
+      the socket excluding the delimiter.
+    - For HTTP-based protocols, the total number of bytes in the HTTP body, as
+      represented by the `Content-Length` header.
+    - For files, the total number of bytes read from the file excluding the
+      delimiter.
+  - `protocol` - The protocol used to send the bytes (i.e., `tcp`, `udp`,
+    `unix`, `http`, `https`, `file`, etc.)
+  - `http_path` - If relevant, the HTTP path, excluding query strings.
+  - `socket` - If relevant, the socket number that bytes were received from.
+- Metrics
+  - MUST increment the `component_received_bytes_total` counter by the defined value with
+    the defined properties as metric tags.
+- Logs
+  - MUST log a `Bytes received.` message at the `trace` level with the
+    defined properties as key-value pairs.
+  - MUST NOT be rate limited.
```

Comment:
Just curious- how do we decide what should and should not be rate limited ?

Reply:
Trace level messages usually aren't rate limited since, if the user is enabling trace messages, they probably want to see every single event.